### PR TITLE
Support aux kernel evaluation of lower-d vars

### LINKED ISSUE #26592: Support aux kernel eval of lower-d vars
## Reason
This is in response to https://github.com/idaholab/moose/discussions/26554. The user wants to populate surface data with a material property that could be computed using higher-d variables. So essentially we need to fill the lower-d var with the face material property evaluation.

## Design
Leverage the `reinitLowerDElem` call in the `ComputeElemAuxBcsThread`

## Impact
More user support

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #26592

### CODE REVIEW

File: framework/include/variables/MooseVariableBase.h [hunk 1]
```diff
@@ -178,6 +178,11 @@ class MooseVariableBase : public MooseObject,
    */
   bool isArray() const { return _is_array; }
 
+  /**
+   * @return whether this variable lives on lower dimensional blocks
+   */
+  bool isLowerD() const { return _is_lower_d; }
```
Comment (GiudGiud) [inline framework/include/variables/MooseVariableBase.h:184]: can a variable be both lowerD and normal D?

File: test/tests/auxkernels/lower_d_var/tests [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  design = 'AuxKernels/index.md'
+  issues = '#26592'
+  [csv]
+    type = CSVDiff
+    input = lower-d-aux-var.i
+    csvdiff = lower-d-aux-var_out.csv
+    requirement = 'The system shall allow populating lower-dimensional auxiliary variables using boundary restricted auxiliary kernels.'
+  []
+[]
```
Comment (GiudGiud) [inline test/tests/auxkernels/lower_d_var/tests:10]: can you please add a h-refined test case?
That would be my first worry on such changes

File: framework/include/variables/MooseVariableBase.h (hunk 1 above)
Comment (lindsayad) [reply framework/include/variables/MooseVariableBase.h:184]: no, that would make evaluation on a boundary face ambiguous

File: test/tests/auxkernels/lower_d_var/tests (hunk 2 above)
Comment (lindsayad) [reply test/tests/auxkernels/lower_d_var/tests:10]: If someone just used `RefineSidesetGenerator`, then we won't have matching refinement of the lower-d elements and the higher-d boundary elements. So your worry is valid ... the code being added here and likely a lot of lower-d element code in MOOSE will fail with refinement

Comment (lindsayad) [reply test/tests/auxkernels/lower_d_var/tests:10]: never mind, I was wrong 😄 we can test this

File: framework/src/variables/MooseVariableBase.C [hunk 3]
```diff
@@ -105,7 +105,8 @@ MooseVariableBase::MooseVariableBase(const InputParameters & parameters)
     _tid(getParam<THREAD_ID>("tid")),
     _count(getParam<unsigned int>("components")),
     _use_dual(getParam<bool>("use_dual")),
-    _is_array(getParam<bool>("array"))
+    _is_array(getParam<bool>("array")),
+    _is_lower_d(blocksMaxDimension() < _mesh.dimension())
```
Comment (GiudGiud) [inline framework/src/variables/MooseVariableBase.C:109]: I m not sure it s that clear cut.

if we have a mesh that has both 2D and 3D regions, completely split not 2D living on the 3D sideset, then a 2D variable would not be lower D necessarily

Comment (lindsayad) [reply framework/src/variables/MooseVariableBase.C:109]: ok I will push an additional commit to see if you like it better

**Review by GiudGiud — APPROVED** [review]

**Comment by lindsayad** [comment]
thanks for another review!

**Comment by lindsayad** [comment]
failure is unrelated

**Pull request merged**